From 23c68e8a8a0f8b3ac894508257c42c0ba6b97052 Mon Sep 17 00:00:00 2001
From: Nell <xianbin.zhu@spacemit.com>
Date: Wed, 27 Nov 2024 20:48:33 +0800
Subject: [PATCH 1314/1448] k1:power-key: fix bug of wakeup count usage of
 power-key

we should use pm_stay_awake in power-key rise interrupt callback and
pm_relax in power-key fall interrupt callback, otherwise the system
may blocked in getting wakep count process; but using pm_wakeup_event
will be more security

Change-Id: Ic73019a3732c5da884523435223a3694619a2469
---
 drivers/input/misc/spacemit-pwrkey.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/input/misc/spacemit-pwrkey.c b/drivers/input/misc/spacemit-pwrkey.c
index ab8616dc2c56..9a33629f909b 100644
--- a/drivers/input/misc/spacemit-pwrkey.c
+++ b/drivers/input/misc/spacemit-pwrkey.c
@@ -30,7 +30,7 @@ static irqreturn_t pwrkey_fall_irq(int irq, void *_pwr)
 		fall_triggered = 1;
 	}
 
-	pm_stay_awake(pwr->dev.parent);
+	pm_wakeup_event(pwr->dev.parent, 0);
 
 	spin_unlock_irqrestore(&pm_lock, flags);
 
@@ -50,7 +50,7 @@ static irqreturn_t pwrkey_rise_irq(int irq, void *_pwr)
 		fall_triggered = 0;
 	}
 
-	pm_relax(pwr->dev.parent);
+	pm_wakeup_event(pwr->dev.parent, 0);
 
 	spin_unlock_irqrestore(&pm_lock, flags);
 
-- 
2.47.0

